File: problems/1/TwoSum.js
/**
 * Created by lillard on 11/9/17.
 */

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numMap = {};
    for (let i = 0; i < nums.length; i++) {
        const targetNum = target - nums[i];
        if (numMap[targetNum] != null && numMap[targetNum] != undefined) {
            return [numMap[targetNum], i];
        }

        numMap[nums[i]] = i;
    }

    return null
};

console.log(twoSum([2, 7, 11, 15], 9));

File: problems/1/TwoSum.go
/**
 * Created by lillard on 11/9/17.
 */

package main

import (
	"fmt"
)

func twoSum(nums []int, target int) (res []int) {
	res = make([]int, 0)
	numMap := make(map[int]int)
	for i, v := range nums {
		targetNum := target - v
		if index, ok := numMap[targetNum]; ok {
			res = append(res, index)
			res = append(res, i)
			break
		}
		numMap[v] = i
	}

	return
}

func main() {
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 9))
}
